### PR TITLE
Fix kubelet restart on the master nodes

### PR DESCRIPTION
**Pull request by alex-vmw**
Fix kubelet restart on the master nodes.

### CODE REVIEW

File: handlers/main.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Copyright 2015 VMware, Inc.  All rights reserved.
 # SPDX-License-Identifier: MIT OR GPL-3.0-only
 ---
+- name: reset failed kubelet
+  become: true
+  command: systemctl reset-failed kubelet
+  notify:
+    - restart kubelet
```
Comment (naveens) [inline handlers/main.yml:8]: Handlers notifying other handlers, is that a documented feature? I believe the conventional way to run multiple handler tasks is to `notify` both of them (`notify` takes a list of handlers, and handlers are run in the order they are defined and not in the order they are notified). Or using the `listen` directive.

File: tasks/main.yml [hunk 2]
```diff
@@ -164,6 +164,14 @@
   become: yes
   template: src=kube-scheduler.yaml dest={{kube_manifests_dir}}/kube-scheduler.yaml
 
+# By the time we get here, kubelet would have been restarting for a while and
+# would have already failed due to the systemd start rate limiting. Since we
+# are specifically waiting below for the apiserver to come up, which requires
+# kubelet, we must force handler execution immediatelly instead of waiting
```
Comment (naveens) [inline tasks/main.yml]: nit: immediately

File: tasks/main.yml [hunk 3]
```diff
@@ -164,6 +164,14 @@
...
+# for handler execution to happen at the end of the play, so that kubelet
+# gets successfully started.
+- meta: flush_handlers
+
 - name: wait for apiserver to be up
   command: curl -i --silent http://localhost:8080
```
Comment (naveens) [inline tasks/main.yml:176]: While we are here, can we switch to using the `uri` module instead of `curl`?

File: handlers/main.yml (hunk 1 above)
Comment (alex-vmw) [reply handlers/main.yml:8]: Hi @naveens,  I am just doing exactly what was done before me for kubelet restart in other roles. Please, look at handlers for the kubelet role or kubernetes-minion role for example.

File: tasks/main.yml (hunk 3 above)
Comment (alex-vmw) [reply tasks/main.yml:176]: Hi @naveens, while I can change this to uri easily I don't see a point of changing code that has worked for many years and is expected to be deprecated within next 6 months. I also checked and the new code base already uses uri module for the same check, so we are not carrying this issue further.

File: tasks/main.yml (hunk 2 above)
Comment (alex-vmw) [reply tasks/main.yml]: Fixed.

**Review by naveens — APPROVED** [review]

**Pull request merged**